transaction.sql

```diff
@@ -0,0 +1,37 @@
+
+BEGIN;
+
+UPDATE animals SET species='unspecified';
+
+ROLLBACK;
+
+
+BEGIN;
+
+UPDATE animals SET species='digimon' WHERE name LIKE '%mon';
+
+UPDATE animals SET species='pokemon' WHERE species is NULL;
+
+COMMIT;
+
+
+BEGIN;
+
+DELETE FROM animals;
+
+ROLLBACK;
+
+
+BEGIN;
+
+DELETE FROM animals WHERE date_of_birth >'2022-1-1';
+
+SAVEPOINT remove_data;
+
+UPDATE animals SET weight_kg= -1*weight_kg;
+
+ROLLBACK TO remove_data;
+
+UPDATE animals SET weight_kg= -1*weight_kg WHERE weight_kg < 0;
+
+COMMIT;
```

Comment:
- [ ] <b>REQUIRED:</b> The project requires to put this code in `queries.sql`. and to make sure that you verify that the change are made before and after COMMIT. something like this:

```sql
   BEGIN;
   UPDATE animals SET species = 'digimon' WHERE name LIKE '%mon'; 
   UPDATE animals SET species = 'pokemon' WHERE species IS NULL; 

   SELECT species from animals; -- verify that change was made
   COMMIT;
   SELECT species from animals; -- verify that change persists after commit
```